### PR TITLE
This push fixed 2 errors enumerated below.

### PR DESCRIPTION
**Pull request by maxpiva**
* Fixed EnumerateDirectories hanging on network folders, used by Rename/Move to delete Empty Directories.

* Fixed Ogg Files Duration not recognized, if MediaInfo does not give a duration, the MediaInformation is lost, and missing files are always added at startup for retrieval, and give errors. Grabbed the Ogg Parser capability from AvDump, and now Ogg Files are correctly timed.

### CODE REVIEW

File: Shoko.Server/Utilities/AVDumpHelper.cs
Comment (revam) [inline Shoko.Server/Utilities/AVDumpHelper.cs:1]: until AVD3 is updated then the URL should not be included in the sauce code, and setting `--ForwardConsoleCursorOnly` breaks the progress parsing.

**Review by revam — CHANGES_REQUESTED** [review]

File: Shoko.Server/Utilities/AVDumpHelper.cs
Comment (maxpiva) [reply Shoko.Server/Utilities/AVDumpHelper.cs:1]: rolledback

**Comment by revam** [comment]
I'll let @da3dsoul review the other changes.

File: Shoko.Server/Services/Ogg/OggFile.cs [hunk 1]
```diff
@@ -0,0 +1,463 @@
+using System;
```
Comment (Cazzar) [inline Shoko.Server/Services/Ogg/OggFile.cs:1]: Would it be worth reformatting this to match with the `.editorconfig` setup?

File: Shoko.Server/Services/VideoLocal_PlaceService.cs [hunk 2]
```diff
@@ -252,8 +253,8 @@ public async Task<RelocationResult> DirectlyRelocateFile(SVR_VideoLocal_Place pl
                 _logger.LogWarning("The existing file at the new location does not have AniDB info. Not moving.");
                 return new()
                 {
-                    Success = false,
-                    ShouldRetry = false,
+                    Success = false, 
+                    ShouldRetry = false,
```
Comment (Cazzar) [inline Shoko.Server/Services/VideoLocal_PlaceService.cs:257]: Formatting - why is spaces getting added here - is your IDE formatter causing this?

File: Shoko.Server/Services/VideoLocal_PlaceService.cs [hunk 3]
```diff
@@ -264,8 +265,8 @@ public async Task<RelocationResult> DirectlyRelocateFile(SVR_VideoLocal_Place pl
                 _logger.LogWarning("The file does not have AniDB info. Not moving.");
                 return new()
                 {
-                    Success = false,
-                    ShouldRetry = false,
+                    Success = false, 
+                    ShouldRetry = false,
```
Comment (Cazzar) [inline Shoko.Server/Services/VideoLocal_PlaceService.cs:269]: Formatting - why is spaces getting added here - is your IDE formatter causing this?

File: Shoko.Server/Services/VideoLocal_PlaceService.cs [hunk 4]
```diff
@@ -834,6 +861,10 @@ public bool RefreshMediaInfo(SVR_VideoLocal_Place place)
 
                 var name = place.FullServerPath.Replace("/", $"{Path.DirectorySeparatorChar}");
                 m = Utilities.MediaInfoLib.MediaInfo.GetMediaInfo(name); // MediaInfo should have libcurl.dll for http
+                if (m?.GeneralStream!=null && m.GeneralStream.Duration == 0 && m.GeneralStream.Format!=null && m.GeneralStream.Format.ToLowerInvariant()=="ogg")
+                {
+                    m.GeneralStream.Duration = CalculateDurationOggFile(name);
+                }
```
Comment (Cazzar) [inline Shoko.Server/Services/VideoLocal_PlaceService.cs:867]:
```suggestion

                if (m?.GeneralStream != null && m.GeneralStream.Duration == 0 && m.GeneralStream.Format != null && m.GeneralStream.Format.ToLowerInvariant() == "ogg")
                {
                    m.GeneralStream.Duration = CalculateDurationOggFile(name);
                }

```

Helps readability.

File: Shoko.Server/Utilities/AVDumpHelper.cs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-﻿#nullable enable
+#nullable enable
```
Comment (Cazzar) [inline Shoko.Server/Utilities/AVDumpHelper.cs:1]: IDE config? Did you change the file format of this? (this almost seems like a change from/to UTF-8 with BOM)

File: Shoko.Server/Utilities/MediaInfoLib/MediaInfo.cs [hunk 6]
```diff
@@ -62,7 +62,7 @@ private static MediaContainer GetMediaInfo_Internal(string filename)
 
             // assuming json, as it starts with {
             var m = JsonConvert.DeserializeObject<MediaContainer>(output, settings);
-            if (m == null)
+            if (m == null || m.media==null)
```
Comment (Cazzar) [inline Shoko.Server/Utilities/MediaInfoLib/MediaInfo.cs:65]:
```suggestion
            if (m == null || m.media == null)
```

**Review by Cazzar — CHANGES_REQUESTED** [review]

File: Shoko.Server/Services/Ogg/OggFile.cs (hunk 1 above)
Comment (maxpiva) [reply Shoko.Server/Services/Ogg/OggFile.cs:1]: Done. Used Format Document from Visual Studio.

File: Shoko.Server/Services/VideoLocal_PlaceService.cs (hunk 2 above)
Comment (maxpiva) [reply Shoko.Server/Services/VideoLocal_PlaceService.cs]: Done

File: Shoko.Server/Services/VideoLocal_PlaceService.cs (hunk 3 above)
Comment (maxpiva) [reply Shoko.Server/Services/VideoLocal_PlaceService.cs]: Done

File: Shoko.Server/Utilities/AVDumpHelper.cs (hunk 5 above)
Comment (maxpiva) [reply Shoko.Server/Utilities/AVDumpHelper.cs:1]: Indeed BOM was removed from the file.

Comment (maxpiva) [reply Shoko.Server/Utilities/AVDumpHelper.cs]: Included BOM back.

File: Shoko.Server/Utilities/MediaInfoLib/MediaInfo.cs (hunk 6 above)
Comment (maxpiva) [reply Shoko.Server/Utilities/MediaInfoLib/MediaInfo.cs]: DONE

**Pull request merged**